api/middleware/synchronize.js

```diff
@@ -0,0 +1,66 @@
+const logger = require('../logger')('sync middleware');
+const { cache, modelIndex } = require('./cache');
+
+const synchronizeAll = getSpecifics =>
+  cache(
+    ['sync-all', modelIndex(getSpecifics)],
```

Comment:
`modelIndex` might not be the best name for this function anymore - it just turns a non-primitive into a number for cacheing purposes.  🤷‍♀️